dockerfiles/Dockerfile.router

```diff
@@ -63,7 +63,7 @@ RUN \
 \nset -e \
 \n \
 \nif [ -f "/usr/bin/heaptrack" ]; then \
-\n    exec heaptrack -o /dist/data/router_heaptrack /dist/router "$@" \
+\n    exec heaptrack -o /dist/data/router_heaptrack/${hostname} /dist/router "$@" \
```

Comment:
This looks to be reading a variable called hostname which I don't think is a standard on linux. You probably mean to use `$(hostname)`. Also, the last element of the path is the actual filename of the file, so I think you have your change in the wrong place in the path. With your change `router_heaptrack` is added as a directory in the path and the filename of the output file changes from a fixed name `router_heaptrack` to a variable name (the hostname).

Has this change been tested in your environment?

I think your change should be:

```
\n exec heaptrack -o /dist/data/$(hostname)/router_heaptrack.$(hostname)  /dist/router "$@" \
```
This tags the hostname as part of the filename. I'm using a `.` as a separator, it could equally well be a `_`.

or:

```
\n exec heaptrack -o /dist/data/$(hostname)/router_heaptrack  /dist/router "$@" \
```
This produces a file with the same name as currently, under a new sub-directory which is the hostname of the executing system. Alternatively, it could be incorporated into the file name as follows:

Note: `heaptrack` appears to provide a %h variable which looks like it would do what you are looking for. However, I've found that to be a bit unreliable, so using `$(hostname)` is safer...

Reply:
Thank you so much for the review, much appreciated, will update as soon as I get some time.